File: tarea-3/etapa-2/Medicina.cpp
#include "../headers/Mascota.h"
#include "../headers/Medicina.h"
#include <iostream>

using std::string;
using std::cout;
using std::endl;


Medicina::Medicina(int id, string nombre, int cantidad) : Item(id, "Medicina", nombre, cantidad) {}

void Medicina::usar(Mascota &mascota) {
    cout << "Usando medicina: " << getNombre() << " en la mascota\n" << endl;

    mascota.salud += 40;
    if (mascota.salud > MAX_STATS) {
        mascota.salud = MAX_STATS;
    }

    cantidad --;
}

File: proyecto/Medicina.py
from Item import Item

class Medicina(Item):

    def __init__(self, id, nombre, cantidad,imagen_path=None):
        super().__init__(id, "Medicina", nombre, cantidad)

    def usar(self, mascota):
        if self._cantidad > 0:
            self._cantidad -= 1
        else:
            print(f"No hay más {self._nombre} disponibles.")
